src/errorpage.cc

```diff
@@ -694,8 +694,18 @@ ErrorState::ErrorState(err_type t, Http::StatusCode status, HttpRequest * req, c
     ale = anAle;
 }
 
-ErrorState::ErrorState(HttpRequest * req, HttpReply *errorReply) :
-    ErrorState(ERR_RELAY_REMOTE)
+ErrorState::ErrorState(err_type aType, Http::StatusCode status, char const *anUrl,
+                       Ip::Address &srcAddress, HttpRequest *aRequest,
+                       const AccessLogEntry::Pointer &al) : 
+                       ErrorState(aType, status, aRequest, al)
+{
+    src_addr = srcAddress;
+    if (anUrl)
+        url = xstrdup(anUrl);
```

Comment:
Since this constructor is using another constructor, it is best _not_ to assume that all future versions of that other constructor will leave url nil. Most likely, this `url` info (whatever that is) should be supplied via ALE (with this parameter removed and the code using it refactored accordingly). If always supplying `url` info via ALE is impossible for some valid reason, then this code should either free the old `url` value (if any) or skip the assignment if `url` happens to be not nil.

Reply:
Leaving a TODO to consolidate constructors onto a single one

Reply:
IMO if the sub-constructor set the url but this one was explicitly used to set it. Then this code SHOULD override the sub-constructor action in exactly the conditional way done.

Reply:
> Alex: this code should either free the old url value (if any) or skip the assignment

> Amos: this code SHOULD override the sub-constructor action

@yadij, it sounds like you think that "free the old url value" is the right choice here. I am not going to argue (at least not until the meanings of these URLs become clear). The code needs changing regardless of the choice.

And again, the best change here is to remove (or, to be precise, undo adding) this second constructor and close this Pandora Box with it.